### PR TITLE
OnDemand TV Page Routing 

### LINKED ISSUE #6569: Re-enable TV Routing - Simorgh
**Is your feature request related to a problem? Please describe.**

The routing was removed while refactoring the radio routing. Adding it back in now.

**Describe the solution you'd like**
Replace the tv routing that was removed during the refactor. The refactor allowed the regex matching to be separated to tv regex and radio regex out for simplicity.

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by rebeccamcginn**
Resolves #6569
Related https://github.com/bbc/simorgh/issues/6629

**Overall change:**

TV and Radio used the same regex matching. I have separated them out into two routes & regex matchers. A previous PR was merged for the radio regex and route; this PR is for re-introducing the TV regex and route. To test this, I have added a simple page which forms the basis of the future onDemand TV Brand page.

**Code changes:**

- Added new route for the future onDemand TV page
- Added basic page to test the routing

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/routes/index.js [hunk 1]
```diff
@@ -15,6 +16,7 @@ export default [
   liveRadio,
   mostRead,
   onDemandRadio,
+  onDemandTV,
```
Comment (ryanmccombe) [inline src/app/routes/index.js:19]: Needs tests in routes/index.test.js 🙂

File: src/server/index.jsx [hunk 2]
```diff
@@ -191,6 +192,19 @@ if (process.env.SIMORGH_APP_ENV === 'local') {
 
       sendDataFile(res, `${dataFilePath}.json`, next);
     })
+    .get(onDemandTvDataPath, async ({ params }, res, next) => {
```
Comment (ryanmccombe) [inline src/server/index.jsx:195]: Needs tests in server/index.test.js 🙂

File: src/app/pages/OnDemandTvPage/index.stories.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
```
Comment (ryanmccombe) [inline src/app/pages/OnDemandTvPage/index.stories.jsx:1]: This is great! storybook is usually forgotten 😢

File: src/app/pages/OnDemandTvPage/index.jsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import React, { useContext } from 'react';
+import styled from 'styled-components';
+import { shape, string, number } from 'prop-types';
+import MetadataContainer from '../../containers/Metadata';
+import { GelPageGrid } from '#app/components/Grid';
+import { ServiceContext } from '../../contexts/ServiceContext';
+
+const getGroups = (zero, one, two, three, four, five) => ({
```
Comment (ryanmccombe) [inline src/app/pages/OnDemandTvPage/index.jsx:8]: Out of scope for this,  but the Gel Grid API should probably just support this use case better - seen this code a few times now

Something like `<GelGrid columns={[6, 6, 6, 6, 8, 20]}>`

File: src/app/routes/onDemandTV/getInitialData/index.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
+import path from 'ramda/src/path';
```
Comment (ryanmccombe) [inline src/app/routes/onDemandTV/getInitialData/index.js:1]: A lot of this is duplicated with the Radio brand - I'd recommend either:

- Removing all the unused stuff from here, and adding it back in as we need it when we're building out the page
- Or leaving it like this, but having both TV and Radio pages use the same `getInitialData` function for now (if possible)

File: src/app/pages/OnDemandTvPage/index.jsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
+import React, { useContext } from 'react';
```
Comment (ryanmccombe) [inline src/app/pages/OnDemandTvPage/index.jsx:1]: Is there anything on this page that is staying, and could be tested?  eg we would usually scaffold the `OnDemandTVPage/index.test.js` file at the same time we add the page

File: src/app/routes/onDemandTV/getInitialData/index.js (hunk 5 above)
Comment (rebeccamcginn) [reply src/app/routes/onDemandTV/getInitialData/index.js:1]: Yeah my plan was to use the same function for both but I don't yet know what data is required for onDemand TV. I'll strip it back and we can build it back up when we come to adding components to the page.

File: src/app/pages/OnDemandTvPage/index.jsx (hunk 6 above)
Comment (rebeccamcginn) [reply src/app/pages/OnDemandTvPage/index.jsx:1]: I tried to build up the tests but there is nothing on the page. I might add back in a title just to ensure the test file is ready to go.

File: src/server/index.jsx (hunk 2 above)
Comment (rebeccamcginn) [reply src/server/index.jsx:195]: Adding now :)

File: src/app/routes/index.js (hunk 1 above)
Comment (rebeccamcginn) [reply src/app/routes/index.js:19]: Nice spot, missed this one

Comment (rebeccamcginn) [reply src/app/routes/index.js:19]: Added

File: src/app/pages/OnDemandTvPage/index.jsx (hunk 6 above)
Comment (rebeccamcginn) [reply src/app/pages/OnDemandTvPage/index.jsx:1]: Brand title is in as is the test file with its sole test

File: src/app/pages/OnDemandTvPage/index.jsx (hunk 4 above)
Comment (rebeccamcginn) [reply src/app/pages/OnDemandTvPage/index.jsx:8]: Agreed. I'll add a note to the board

**Review by ryanmccombe — APPROVED** [review]
party

**Review by jroebu14 — APPROVED** [review]

**Pull request merged**